### PR TITLE
SDK-230 -- Unwind SystemObserver

### PR DESCRIPTION
**Pull request by apeterson-branch**
## Reference
SDK-230 -- Unwind SystemObserver

## Description
SystemObserver.java is used to collect things like advertising IDs, Hardware IDs, etc. The usage of this class is a bit convoluted, and because of statics and copies of these classes it isn't always working as expected.

For example, calls through DeviceInfo.java will only use the very first SystemObserver (because of a static), where other calls to a SystemObserver can be made returning completely different results. "disableDeviceIDFetch_" is one such variable, that depending on which direction a call is made from, will set different values in a payload.

## Testing Instructions
* Sample app should continue to function.
* Unit tests should all pass

## Notes
Unwinding this was a little more than I had hoped when first undertaking this task.  I essentially removed the ability to construct a SystemObserver outside of DeviceInfo.

## Risk Assessment [`MEDIUM`]
The code changes affected a lot of areas, and a lot of internal method signatures.  While the behavior might be slightly different when moving in and out of debug, test, or simulation mode -- it should now be internally consistent regardless of where the call is coming from.

- [x] I, the PR creator, have tested — integration, unit, or otherwise — this code.

## Reviewer Checklist (To be checked off by the reviewer only)

- [ ] JIRA Ticket is referenced in MR title
- Correctness & Style
    - [ ] Conforms to [Style Guides](https://google.github.io/styleguide/javaguide.html)
    - [ ] Mission critical pieces are documented in code and out of code as needed
- [ ] Unit Tests reviewed and test issue sufficiently
- [ ] Functionality was reviewed in QA independently by another engineer on the team (clone for each required reviewer)

### CODE REVIEW

File: Branch-SDK/src/io/branch/referral/BranchStrongMatchHelper.java [hunk 1]
```diff
@@ -25,26 +25,26 @@
...
-    boolean isStrongMatchUrlLaunched = false;
+    private boolean isStrongMatchUrlLaunched = false;
 
-    Class<?> CustomTabsClientClass;
-    Class<?> CustomServiceTabConnectionClass;
-    Class<?> CustomTabsCallbackClass;
-    Class<?> CustomTabsSessionClass;
-    Class<?> ICustomTabsServiceClass;
+    private Class<?> CustomTabsClientClass;
+    // private Class<?> CustomServiceTabConnectionClass;
```
Comment (aaaronlopez) [inline Branch-SDK/src/io/branch/referral/BranchStrongMatchHelper.java]: remove

File: Branch-SDK/src/io/branch/referral/DeviceInfo.java [hunk 2]
```diff
@@ -234,19 +194,28 @@ public String getPackageName() {
      * @return {@link String} with app version value
      */
     public String getAppVersion() {
-        return appVersion_;
+        return systemObserver_.getAppVersion(context_);
     }
 
-    public boolean isHardwareIDReal() {
```
Comment (aaaronlopez) [inline Branch-SDK/src/io/branch/referral/DeviceInfo.java]: When we remove this, let's ensure the key/value is still sent on outgoing events in request payload

File: Branch-SDK/src/io/branch/referral/BranchStrongMatchHelper.java (hunk 1 above)
Comment (apeterson-branch) [reply Branch-SDK/src/io/branch/referral/BranchStrongMatchHelper.java]: Removed.

**Comment by aaaronlopez** [comment]
let's do it! 👍

**Pull request merged**